### ISSUE TITLE
Missing card flow listener - 0.8.1

### ISSUE THREAD

**Issue by shaarkys** [lvanderree/com.synplyworks.thenmore#25]
Hi, 
I saw you are closing open issues for SDK3 - maybe not sure if you have not forgotten on this issue - it forced me to use still the previous SDK2 version.

Thank you.

https://community.homey.app/t/timer-0-8-1-test-homey-v3-0-7-3-beta-homey-v2-v0-4-3-stable-homey-v1-5-fka-then-more-add-timers-to-temporarily-turn-on-devices/3722/76?u=sharkys

**Comment by shaarkys** [comment]
Hi @lvanderree , may I ask you please if this will be fixed or there is some issue that prevents it (I mean in the SDK)...a little bit worried it will be not working after SDK3 switch. Thank you.

**Comment by lvanderree** [comment]
hi @shaarkys 
unfortunately I haven't got time to make the change yet. 
It will be next thing if I find a moment to take a look the the timer app

**Comment by shaarkys** [comment]
Thank you - no problem, I just thought it's like you have almost reached finish line ;-)

![obrazek](https://user-images.githubusercontent.com/14859895/225871364-d3b4e5f9-0f71-45a7-a09a-b102f70a15a1.png)

**Comment by lvanderree** [comment]
I cannot test it myself at the moment, but please try this https://homey.app/nl-nl/app/com.synplyworks.thenmore/Timer/test/

**Comment by shaarkys** [comment]
Thank you, tested and it's crashing right after start. Unfortunately the Android Homey beta app is not showing any details of the crash - caa63f20-37ee-47bb-8801-001b0ef66b27

**Comment by lvanderree** [comment]
Ahh too bad, I think I fixed the issue, but also converted the code to typescript. So probably I introduced a small but fatal issue there. 

The only problem was I cannot run the app myself at the moment, while I am not at home and my VPN does not allow me to deploy to homey. So please wait till I get home again so I do some testing myself

**Comment by shaarkys** [comment]
No problem at all... If it helps, this is the output of the crash : 


```
Unable to initialize app Error: Cannot find module 'source-map-support'
Require stack:
- /app.js
- /opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Util.js
- /opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Device.js
- /opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/homey.js
- /opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/index.js
- /opt/homey-client/system/manager/ManagerApps/AppProcess/bootstrap.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at Module.require (/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/index.js:1:183)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/app.js:6:46)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app.js',
    '/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Util.js',
    '/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Device.js',
    '/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/homey.js',
    '/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/index.js',
    '/opt/homey-client/system/manager/ManagerApps/AppProcess/bootstrap.js'
  ]
}
/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Homey.js:1
"use strict";const path=require("path"),SimpleClass=require("./SimpleClass.js"),Manager=require("./Manager.js");class Homey extends SimpleClass{constructor(e,t,s,i,r){super(),this.dir=e,this.tmpdir=path.join(e,"tmp"),this.version=s,this.platform=i,this.platformVersion=r,this.env={},this.manifes
t=t,this.app={},this.__readyPromise=new Promise(e=>{this.__readyResolve=e}),this._timers=new Set,this._destroyed=!1,this.apps={},this.arp={},this.audio={},this.ble={},this.cloud={},this.clock={},this.drivers={},this.discovery={},this.flow={},this.geolocation={},this.i18n={},this.images={},this.
insights={},this.ledring={},this.nfc={},this.notifications={},this.rf={},this.settings={},this.speechInput={},this.speechOutput={},this.zigbee={},this.zwave={},this.api={}}ready(){return this.__readyPromise}markReady(){this.__readyResolve()}hasPermission(e){return!1!==Array.isArray(this.manifes
t.permissions)&&this.manifest.permissions.includes(e)}__(e,t){}setTimeout(e,t,...s){if(this._destroyed)return null;const i=setTimeout(()=>{this._timers.delete(i),e(...s)},t);return this._timers.add(i),i}clearTimeout(e){clearTimeout(e),this._timers.delete(e)}setInterval(e,t,...s){if(this._destro
yed)return null;const i=setInterval(e,t,...s);return this._timers.add(i),i}clearInterval(e){clearInterval(e),this._timers.delete(e)}destroy(){this.__debug("Homey Destroyed"),this.emit("unload"),this._destroyed=!0;for(const e of this._timers.values())clearTimeout(e);const e={emit:async e=>{throw
 this.__debug(`Emit after destroy ${e}`),new Error(`Cannot send "${e}" because the app instance has been destroyed.\n`+"This may indicate that your app is not cleaning up all resources in `onUninit()`.")}},t=this.app,s=this.drivers;for(const t of Object.keys(this)){const s=this[t];s instanceof
Manager&&(this[t]=void 0,s.homey=void 0,s.__client=e,s.removeAllListeners(),Object.defineProperty(this,t,{get(){throw new Error(`Cannot access \`this.homey.${t}\` because the app instance has been destroyed.\n`+"This may indicate that your app is not cleaning up all resources in `onUninit()`.")
}}))}this.app=void 0,Object.defineProperty(this,"app",{get(){throw new Error("Cannot access `this.homey.app` because the app instance has been destroyed.\nThis may indicate that your app is not cleaning up all resources in `onUninit()`.")}}),this.removeAllListeners(),Promise.resolve().then(()=>
t.onUninit()).catch(this.error);for(const e of Object.values(s.getDrivers())){Promise.resolve().then(()=>e.onUninit()).catch(this.error);for(const t of e.getDevices())Promise.resolve().then(()=>t.onUninit()).catch(this.error)}}}module.exports=Homey;





Error: Cannot access `this.homey.app` because the app instance has been destroyed.
This may indicate that your app is not cleaning up all resources in `onUninit()`.
    at Homey.get [as app] (/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Homey.js:1:2133)
    at Homey.destroy (/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/Homey.js:1:1689)
    at IPCSocket.<anonymous> (/opt/homey-client/system/manager/ManagerApps/AppProcess/node_modules/homey-apps-sdk-v3/lib/SDK.js:124:43)
    at Object.onceWrapper (node:events:645:28)
    at IPCSocket.emit (node:events:538:35)
    at process.<anonymous> (/opt/homey-client/system/lib/IPCSocket.js:1:395)
    at process.emit (node:events:538:35)
    at finish (node:internal/child_process:922:14)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)

--- INFO: com.synplyworks.thenmore has been killed ---

```

**Comment by shaarkys** [comment]
Changed in package.json : 
from  
`"@types/source-map-support": "^0.5.6", `
to
`"@types/source-map-support": "",`
then npm uninstall source-map-support and npm install source-map-support and it's running fine.

Yes, naturally I don't know what I'm doing but it works ;-) 

However the original error has not disappeared, actually changed to new one : **Callback is not a function**

**Comment by lvanderree** [comment]
super, that should both be fixed then in 0.8.3 

it is here to test again https://homey.app/a/com.synplyworks.thenmore/test/

**Comment by shaarkys** [comment]
It works ! ;-) THANK YOU.

![obrazek](https://user-images.githubusercontent.com/14859895/226066822-920f452d-d6e6-4152-8140-93704e03e0e4.png)
